token-lending/program/src/instruction.rs

```diff
@@ -914,3 +953,43 @@ pub fn set_lending_market_owner(
         data: LendingInstruction::SetLendingMarketOwner { new_owner }.pack(),
     }
 }
+
+/// Creates a `FlashLoan` instruction.
+#[allow(clippy::too_many_arguments)]
+pub fn flash_loan(
+    program_id: Pubkey,
+    destination_liquidity_pubkey: Pubkey,
+    reserve_pubkey: Pubkey,
+    reserve_liquidity_pubkey: Pubkey,
+    lending_market_pubkey: Pubkey,
+    derived_lending_market_authority: Pubkey,
+    flash_loan_recevier_pubkey: Pubkey,
+    flash_loan_receiver_program_derived_account: Pubkey,
+    flash_loan_fees_account_info: Pubkey,
+    host_fee_recipient: Pubkey,
+    amount: u64,
+    additional_params: Vec<Pubkey>,
+) -> Instruction {
+    let mut accounts = vec![
+        AccountMeta::new(destination_liquidity_pubkey, false),
+        AccountMeta::new_readonly(reserve_pubkey, false),
+        AccountMeta::new(reserve_liquidity_pubkey, false),
+        AccountMeta::new_readonly(lending_market_pubkey, false),
+        AccountMeta::new_readonly(derived_lending_market_authority, false),
+        AccountMeta::new_readonly(flash_loan_recevier_pubkey, false),
+        AccountMeta::new_readonly(flash_loan_receiver_program_derived_account, false),
+        AccountMeta::new_readonly(spl_token::id(), false),
+        AccountMeta::new(flash_loan_fees_account_info, false),
+        AccountMeta::new(host_fee_recipient, false),
+    ];
+    accounts.extend(
+        additional_params
+            .into_iter()
+            .map(|additional_param| AccountMeta::new(additional_param, false)),
```

Comment:
Should all these accounts be writable? It seems like that should be up to the caller who knows what accounts are expected by the receiving program.

Reply:
Fixed as suggested below.

Reply:
Sorry, that was a mistake. Fixed now.